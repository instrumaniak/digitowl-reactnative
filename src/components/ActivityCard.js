/**
 *   Card component used in Activity screen
 */

import React from 'react'

import {
  View,
  Text,
  StyleSheet
} from 'react-native'

import {
  Icon
} from 'native-base'

import { human } from 'react-native-typography'
import { colors } from '../config'

const ActivityCard = () => (
  <View style={styles.container}>
    <View style={styles.textContainer}>
      <Text style={styles.text} >Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam dictum ante vitae tortor vulputate, at posuere ante lacinia.</Text>
      <Text style={styles.time}>3 min. ago, 1:45 PM</Text>
    </View>

    <View style={styles.iconContainer}>
      <Icon name='ios-arrow-forward' style={styles.icon}/>
    </View>
  </View>
)

const styles = StyleSheet.create({
  container: {
    padding: 10,
    marginBottom: 5,
    backgroundColor: 'white',
    elevation: 2,
    borderRadius: 2,
    flexDirection: 'row'
  },
  textContainer: {
    flex: 15,
    paddingLeft: 2
  },
  text: {
    ...human.bodyObject,
    color: colors.grey1,
    marginBottom: 10
  },
  time: {
    ...human.subheadObject,
    color: colors.grey,
  },
  iconContainer: {
    justifyContent: 'center',
    alignItems: 'flex-end',
    flex: 1
  },
  icon: {
    color: colors.grey
  }
})

export default ActivityCard
