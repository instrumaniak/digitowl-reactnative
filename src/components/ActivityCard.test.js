/**
 *   Tests for component: ActivityCard
 */

import React from 'react'
import ActivityCard from './ActivityCard'

import renderer from 'react-test-renderer'

test('Renders correctly', () => {
  const tree = renderer.create(<ActivityCard />).toJSON()
  expect(tree).toMatchSnapshot()
})
