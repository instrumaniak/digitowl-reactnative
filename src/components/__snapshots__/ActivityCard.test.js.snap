// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 2,
      "elevation": 2,
      "flexDirection": "row",
      "marginBottom": 5,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 15,
        "paddingLeft": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#5B6267",
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "400",
          "letterSpacing": -0.408,
          "lineHeight": 22,
          "marginBottom": 10,
        }
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam dictum ante vitae tortor vulputate, at posuere ante lacinia.
    </Text>
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#c3c6c7",
          "fontFamily": "System",
          "fontSize": 15,
          "fontWeight": "400",
          "letterSpacing": -0.24,
          "lineHeight": 20,
        }
      }
    >
      3 min. ago, 1:45 PM
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#c3c6c7",
            },
          ],
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      Ôèë
    </Text>
  </View>
</View>
`;
